nilearn/connectivity/connectivity_matrices.py

```diff
@@ -0,0 +1,349 @@
+                                          # the tangent space at point gmean
+
+        # Update of the minimizer
+        vals_log, vecs_log = linalg.eigh(logs_mean)
+        gmean_sqrt = _form_symmetric(np.sqrt, vals_gmean, vecs_gmean)
+        # Move along the geodesic
+        gmean = gmean_sqrt.dot(
+            _form_symmetric(np.exp, vals_log * step, vecs_log)).dot(gmean_sqrt)
+
+        # Update the norm and the step size
+        if norm < norm_old:
+            norm_old = norm
+        elif norm > norm_old:
+            step = step / 2.
+            norm = norm_old
+        if tol is not None and norm / gmean.size < tol:
+            break
+    if tol is not None and norm / gmean.size >= tol:
+        warnings.warn("Maximum number of iterations {0} reached without "
+                      "getting to the requested tolerance level "
+                      "{1}.".format(max_iter, tol))
+
+    return gmean
+
+
+def sym_to_vec(symmetric):
+    """Return the flattened lower triangular part of an array, after
+    multiplying above the diagonal elements by sqrt(2).
+
+    Acts on the last two dimensions of the array if not 2-dimensional.
+
+    Parameters
+    ----------
+    symmetric : numpy.ndarray, shape (..., n_features, n_features)
+        Input array.
+
+    Returns
+    -------
+    output : numpy.ndarray, shape (..., n_features * (n_features + 1) / 2)
+        The output flattened lower triangular part of symmetric.
+    """
+    scaling = sqrt(2) * np.ones(symmetric.shape[-2:])
+    np.fill_diagonal(scaling, 1.)
+    tril_mask = np.tril(np.ones(symmetric.shape[-2:])).astype(np.bool)
+    return symmetric[..., tril_mask] * scaling[tril_mask]
+
+
+def _cov_to_corr(covariance):
+    """Return correlation matrix for a given covariance matrix.
+
+    Parameters
+    ----------
+    covariance : 2D numpy.ndarray
+        The input covariance matrix.
+
+    Returns
+    -------
+    correlation : 2D numpy.ndarray
+        The ouput correlation matrix.
+    """
+    diagonal = np.atleast_2d(1. / np.sqrt(np.diag(covariance)))
+    correlation = covariance * diagonal * diagonal.T
+    return correlation
+
+
+def _prec_to_partial(precision):
+    """Return partial correlation matrix for a given precision matrix.
+
+    Parameters
+    ----------
+    precision : 2D numpy.ndarray
+        The input precision matrix.
+
+    Returns
+    -------
+    partial_correlation : 2D numpy.ndarray
+        The 2D ouput partial correlation matrix.
+    """
+    partial_correlation = -_cov_to_corr(precision)
+    np.fill_diagonal(partial_correlation, 1.)
+    return partial_correlation
+
+
+class ConnectivityMeasure(BaseEstimator, TransformerMixin):
+    """A class that computes different kinds of functional connectivity
+    matrices.
+
+    Parameters
+    ----------
+    cov_estimator : estimator object, optional.
+        The covariance estimator.
+
+    kind : {"correlation", "partial correlation", "robust dispersion",\
```

Comment:
Think of it as a marketting problem, not a mathematical one. People need
to be able to remember the name, and link it to publications, possibly
via Google. In that sense 'affine invariant' is not great, because the
terms hasn't been used in fMRI and EEG.

'tangent' is OK, as I think that it conveys the right idea. But in fMRI,
Bernard has been pushing forward Riemannian, and people in EEG too.
That's was I was thinking of 'manifold'. But I am OK with 'tangent'.